array/DNAm/preprocessing/CETYGOdeconvolution.r

```diff
@@ -0,0 +1,195 @@
+##---------------------------------------------------------------------#
+##
+## Title: Run CETYGO cell type deconvolution
+##
+## Purpose of script: Use CETYGO brain ref panal to run CETYGO on the samples
+##
+## Author: Emma Walker
+##
+## Date Created: Feb 2024
+##
+##---------------------------------------------------------------------#
+
+
+#----------------------------------------------------------------------#
+# NOTES
+#----------------------------------------------------------------------#
+# project folder is provided on command line
+# More info on the CETYGO package can be found here: 
+#   https://github.com/ejh243/CETYGO/wiki/Deconvolution-of-brain-cell-types
+
+
+#----------------------------------------------------------------------#
+# DEFINE PARAMETERS
+#----------------------------------------------------------------------#
+
+args<-commandArgs(trailingOnly = TRUE)
+dataDir <- args[1]
+
+gdsFile <-paste0(dataDir, "/2_gds/raw.gds")
+configFile <- paste0(dataDir, "/config.r")
+
+gdsObj<-ifelse(file.exists(gdsFile), TRUE, ifelse(file.exists(msetFile), FALSE, NA))
+
+source(configFile)
+
+
+#----------------------------------------------------------------------#
+# LOAD PACKAGES
+#----------------------------------------------------------------------#
+
+library(CETYGO)
+
+if(is.na(gdsObj)){
+	message("No data to load")
+} else{
+	if(gdsObj){
+		message("Loading data from gds object")
+		library(bigmelon)
+	} else {
+		if(!gdsObj){
+			message("Loading data from mset object")	
+			library(wateRmelon)
+		}
+}
+}
+
+
+
+#----------------------------------------------------------------------#
+# IMPORT DATA
+#----------------------------------------------------------------------#
+
+setwd(dataDir)
+library(gridExtra)
+
+# load sample sheet
+sampleSheet<-read.csv("0_metadata/sampleSheet.csv", na.strings = c("", "NA"), stringsAsFactors = FALSE)
+# if no column Basename, creates from columns Chip.ID and Chip.Location
+if(!"Basename" %in% colnames(sampleSheet)){
+	sampleSheet$Basename<-paste(sampleSheet$Chip.ID, sampleSheet$Chip.Location, sep = "_")
+}
+
+if(gdsObj){
+
+	gfile<-openfn.gds(gdsFile, readonly = FALSE, allow.fork = TRUE)
+	# ensure sample sheet is in same order as data
+	sampleSheet<-sampleSheet[match(colnames(gfile), sampleSheet$Basename),]
+	# extract a few useful matrices
+	if(toupper(arrayType) == "V2"){
+	  rawbetas<-epicv2clean(betas(gfile)[])
+} else {
+	rawbetas<-betas(gfile)[,]
+}
+    closefn.gds(gfile)
+}else {
+	if(!gdsObj){
+		load(msetFile)
+		# ensure sample sheet is in same order as data
+		sampleSheet<-sampleSheet[match(colnames(msetEPIC), sampleSheet$Basename),]
+		# extract a few useful matrices
+		rawbetas<-betas(msetEPIC)
+}
+}
+
+
+  
+
+
+#----------------------------------------------------------------------#
+# DEFINE CETYGO FUNCTION
+#----------------------------------------------------------------------#
+
+adultBrainCETYGO <- function(betas, cellType){
```

Comment:
Do we have a convention on where we should store function definitions? Would we prefer to put this function in the cetygo package as possibly useful for other?

Reply:
Agreed, will add to CETYGO package